internal/db/filter/parser.go

```diff
@@ -265,11 +272,72 @@ func convertValue(val *Value) interface{} {
 		return strings.ToLower(*val.Boolean) == "true"
 	}
 
+	if val.ValueList != nil {
+		// Convert list of values to slice
+		var values []interface{}
+		for _, singleVal := range val.ValueList.Values {
+			values = append(values, convertSingleValue(singleVal))
+		}
+		return values
+	}
+
 	return nil
 }
 
+func convertSingleValue(val *SingleValue) interface{} {
+	if val.String != nil {
+		return processStringValue(*val.String)
+	}
+
+	if val.Integer != nil {
+		return *val.Integer
+	}
+
+	if val.Float != nil {
+		return *val.Float
+	}
+
+	if val.Boolean != nil {
+		return strings.ToLower(*val.Boolean) == "true"
+	}
+
+	return nil
+}
+
+// processStringValue removes quotes and handles escape sequences
+func processStringValue(str string) string {
+	// Remove quotes from string
+	result := strings.Trim(str, `"'`)
+	// Handle escape sequences
+	result = strings.ReplaceAll(result, `\"`, `"`)
+	result = strings.ReplaceAll(result, `\'`, `'`)
+	result = strings.ReplaceAll(result, `\\`, `\`)
+	return result
+}
+
 // inferValueType determines the appropriate value type based on the actual value
 func inferValueType(val *Value) string {
+	if val.String != nil {
+		return StringValueType
+	}
+	if val.Integer != nil {
+		return IntValueType
+	}
+	if val.Float != nil {
+		return DoubleValueType
+	}
+	if val.Boolean != nil {
+		return BoolValueType
+	}
+	if val.ValueList != nil && len(val.ValueList.Values) > 0 {
+		// For lists, infer type from the first value
+		return inferSingleValueType(val.ValueList.Values[0])
+	}
+	return StringValueType // default to string
```

Comment:
Can this code be simplified to:

```suggestion
	if val.ValueList != nil && len(val.ValueList.Values) > 0 {
		// For lists, infer type from the first value
		return inferSingleValueType(val.ValueList.Values[0])
	}
	return inferSingleValueType(val)
```